### PR TITLE
feat: team

### PR DESCRIPTION
**Pull request by kiki7000**
## About
Added [team](https://discord.com/developers/docs/topics/teams#data-models-team-object) to harmony

## Status

- [x] These changes have been tested against Discord API or do not contain API change.
- [ ] This PR includes only documentation changes, no code change.
- [ ] This PR introduces some Breaking changes.

### CODE REVIEW

File: src/types/team.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+export interface TeamPayload {
+  id: string
+  name: string
+  icon: string
+  owner_user_id: string
+  members: TeamMemberPayload[]
+}
+
+export interface TeamMemberPayload {
+  membership_state: number
```
Comment (DjDeveloperr) [inline src/types/team.ts:12]: Please use enum here, https://discord.com/developers/docs/topics/teams#data-models-membership-state-enum

File: src/structures/application.ts [hunk 2]
```diff
@@ -10,6 +11,7 @@ export class Application extends SnowflakeBase {
   description: string
   summary: string
   bot?: User
+  team: Team | null
```
Comment (DjDeveloperr) [inline src/structures/application.ts:14]: I'd prefer undefined over null here

**Review by DjDeveloperr — CHANGES_REQUESTED** [review]
LGTM 🎉 , some nits

**Comment by kiki7000** [comment]
Resolved

File: src/structures/team.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  members: TeamMember[]
+
+  constructor(client: Client, data: TeamPayload) {
+    super(client, data)
+
+    this.id = data.id
+    this.name = data.name
+    this.icon = data.icon
+    this.members = data.members.map(member => new TeamMember(client, member, this))
+    this.owner = this.members.filter(member => member.id === data.owner_user_id)[0]
```
Comment (DjDeveloperr) [inline src/structures/team.ts:34]: .find would be better here

**Review by DjDeveloperr — CHANGES_REQUESTED** [review]

File: src/structures/application.ts [hunk 4]
```diff
@@ -20,5 +22,6 @@ export class Application extends SnowflakeBase {
     this.description = data.description
     this.summary = data.summary
     this.bot = data.bot !== undefined ? new User(client, data.bot) : undefined
+    this.team = data.team ? new Team(client, data.team) : undefined
```
Comment (DjDeveloperr) [inline src/structures/application.ts:25]: Please fix lint error here so we can land this PR

**Review by DjDeveloperr — CHANGES_REQUESTED** [review]

**Review by DjDeveloperr — APPROVED** [review]
LGTM, thanks!

**Pull request merged**